HW1/P4/P4_bfs.py

```diff
@@ -0,0 +1,32 @@
+from pyspark import AccumulatorParam
+
+class SetAccumulator(AccumulatorParam):
+    def zero(self, initialValue):
+        return set()
+
+    # union sets for visited
+    def addInPlace(self, s1, s2):
+        s1 |= s2
+        return s1
+
+# find the number of reachable nodes from starting character
+def ss_bfs(sc, rdd, character):
+	# initialize variables
+    visited = sc.accumulator(set(), SetAccumulator())
+    frontier = {character}
+    distance = 0
+    
+    # terminate if found nothing new
+    while len(frontier):
+    	# update values
+        distance += 1
+        prev_vals = set(list(visited.value))
+
+        # iterate through current level and update visited 
+        rdd.filter(lambda (x,y): x in frontier)
```

Comment:
Nice use of a set accumulator.  I'm curious how well it performs, because I found set accumulators to be for some reason slow.